service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/jobs/JobClient.java

```diff
@@ -434,21 +580,52 @@ public synchronized JobResult getNextJob(Query query) throws IOException, IotHub
      * @throws IOException If any of the input parameters are incorrect
      * @throws IotHubException If IotHub failed to respond
      */
-    public synchronized Query queryJobResponse(JobType jobType, JobStatus jobStatus, Integer pageSize) throws IOException, IotHubException
+    public synchronized Query queryJobResponse(JobType jobType, JobStatus jobStatus, Integer pageSize)
+        throws IOException, IotHubException
     {
         if (pageSize <= 0)
         {
-            //Codes_SRS_JOBCLIENT_25_042: [If the pageSize is null, zero or negative, the queryJobResponse shall throw IllegalArgumentException.]
             throw new IllegalArgumentException("pagesize cannot be negative or zero");
         }
 
-        //Codes_SRS_JOBCLIENT_25_043: [If the pageSize is not specified, default pageSize of 100 shall be used.] SRS_JOBCLIENT_25_044: [The queryDeviceJob shall create a query object for the type JOB_RESPONSE.]
         Query jobResponseQuery = new Query(pageSize, QueryType.JOB_RESPONSE);
 
-        //Codes_SRS_JOBCLIENT_25_045: [The queryDeviceJob shall send a query request on the query object using Query URL, HTTP GET method and wait for the response by calling sendQueryRequest.]
         String jobTypeString = (jobType == null) ? null : jobType.toString();
         String jobStatusString = (jobStatus == null) ? null : jobStatus.toString();
-        jobResponseQuery.sendQueryRequest(iotHubConnectionString, iotHubConnectionString.getUrlQuery(jobTypeString, jobStatusString), HttpMethod.GET, MAX_TIMEOUT);
+        ProxyOptions proxyOptions = options.getProxyOptions();
+        Proxy proxy = proxyOptions != null ? proxyOptions.getProxy() : null;
+
+        if (this.credential != null)
```

Comment:
curious, can this logic live in the Query class instead?

Reply:
By passing in all three of the possible authentication providers and letting the query class figure it out? Let me try it out and see what that saves us here

Reply:
yes, I saw the same sort of logic a couple of times in this PR.

Reply:
just something to think about, it might not actually pan out.

Reply:
This will help simplify our code, so sure I can do this